apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/TimedBufferingConsumerTest.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using Moq;
+using System;
+using System.Collections.Generic;
+using Xunit;
+
+namespace Google.Cloud.Diagnostics.Common.Tests
+{
+    public class TimedBufferingConsumerTest
+    {
+        private static TimeSpan _waitTime = TimeSpan.FromSeconds(5);
+
+        private DateTime _start = DateTime.UtcNow;
+
+        private TimedBufferingConsumer<int> GetConsumer(IConsumer<int> consumer, IClock clock)
+        {
+            return TimedBufferingConsumer<int>.Create(consumer, _waitTime, clock);
+        }
+
+        /// <summary>
+        /// Creates an <see cref="IClock"/> that returns the given times in order.
```

Comment:
Note that we have a FakeClock - we could potentially add this functionality to that...